### PR TITLE
Lw/surface cb split

### PR DESCRIPTION
**Pull request by aleks01010101**

### CODE REVIEW

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/InputSurfaceCommon.hlsl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return SAMPLE_TEXTURE2D(_MainTex, sampler_MainTex, uv);
+}
+
+half3 Normal(float2 uv)
+{
+#if _NORMALMAP
+    #if BUMP_SCALE_NOT_SUPPORTED
+        return UnpackNormal(SAMPLE_TEXTURE2D(_BumpMap, sampler_BumpMap, uv));
+    #else
+        return UnpackNormalScale(SAMPLE_TEXTURE2D(_BumpMap, sampler_BumpMap, uv), _BumpScale);
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/InputSurfaceCommon.hlsl]: Not something to look at this point, but it seems wrong that common depends on each Surface include file. We could remove the dependency if we pass as argument the constants. (_BumpScale, _EmissionColor, etc) and we rename the functions SampleNormal, SampleEmission, etc. _MainTex_ST and Alpha functions there would need more thought.

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl [hunk 2]
```diff
@@ -214,7 +226,7 @@ half AlphaBlendAndTest(half alpha)
 #else
     half result = 1.0h;
 #endif
-    AlphaDiscard(result, _Cutoff, 0.0001h);
+    AlphaDiscard(result, (half)_Cutoff, 0.0001h);
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl]: isn't _Cutoff already half?

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/ParticlesPBR.hlsl [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#ifndef LIGHTWEIGHT_PARTICLES_PBS_INCLUDED
+#define LIGHTWEIGHT_PARTICLES_PBS_INCLUDED
+
+#include "Particles.hlsl"
+#include "SurfaceData.hlsl"
+
+TEXTURE2D(_MetallicGlossMap);   SAMPLER(sampler_MetallicGlossMap);
+
+CBUFFER_START(UnityPerMaterial_ParticlePBS)
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/ParticlesPBR.hlsl]: I wonder if we have any tradeoff between CB and global. a CB for 2 halfs seems an overshoot at first.

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/SurfaceData.hlsl [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#ifndef LIGHTWEIGHT_SURFACE_DATA_INCLUDED
+#define LIGHTWEIGHT_SURFACE_DATA_INCLUDED
+
+// Must match Lightweigth ShaderGraph master node
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/SurfaceData.hlsl]: Did shader graph templates got updated with this? Or maybe we could add this to InputCommon.hlsl?

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl (hunk 2 above)
Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/Particles.hlsl]: Hmm... It is, but the compiler complained about a possible precision loss

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/SurfaceData.hlsl (hunk 4 above)
Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/SurfaceData.hlsl]: It's included in InputSurfaceCommon

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/ParticlesPBR.hlsl (hunk 3 above)
Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/ShaderLibrary/ParticlesPBR.hlsl]: Perhaps I should just put these two in the common particle CB and be done with it then? :)

**Pull request merged**